core/arch/aarch64/clean_call_opt.c

```diff
@@ -209,11 +227,27 @@ analyze_callee_regs_usage(dcontext_t *dcontext, callee_info_t *ci)
                 ci->num_simd_used++;
             }
         }
+
+        /* NZCV register usage */
+        if (!ci->write_flags && TESTANY(EFLAGS_WRITE_ARITH,
+                instr_get_arith_flags(instr, DR_QUERY_INCLUDE_ALL))) {
+            LOG(THREAD, LOG_CLEANCALL, 2,
+                "CLEANCALL: callee "PFX" updates aflags\n", ci->start);
+            ci->write_flags = true;
+        }
+
+        if (!ci->read_flags && TESTANY(EFLAGS_READ_ARITH,
+                instr_get_arith_flags(instr, DR_QUERY_INCLUDE_ALL))) {
+            LOG(THREAD, LOG_CLEANCALL, 2,
+                "CLEANCALL: callee "PFX" reads aflags from caller\n",
+                ci->start);
+            ci->read_flags = true;
+        }
     }
 
-    /* FIXME i#2796: the following checks are still missing:
-     *    - analysis of eflags (depends on i#2263)
-     */
+    if (ci->read_flags || ci->write_flags) {
```

Comment:
Do we have to save the flags if the callee just reads the flags, but does not write them?

Reply:
The calling convention makes no guarantees NZCV, it states [1]: 

"The N, Z, C and V flags are undefined on entry to and return from a public interface."

Therefore read_flags can be ignored and a slot should not be reserved, I'll remove it.

[1] http://infocenter.arm.com/help/topic/com.arm.doc.ihi0055b/IHI0055B_aapcs64.pdf